File: Project_02_C#/project_2_CSharp/project_2_CSharp/App/FacebookApp.cs
using project_2_CSharp.product;
using System;
using System.Collections.Generic;
using System.Text;

namespace project_2_CSharp.App
{
    class FacebookApp:IAndroidCompatible
    {
        private string appName = "Facebook for android";
        private string version = "12.3";

        public string downloading()
        {
            return "Downloading " + ToString();
        }

        public string OSCompatible
        {
            get {
                return "Android";
            }
        }

        public override string ToString()
        {
            return "app name: " + appName  + ", Version: " + version;
        }

        public bool check(string os)
        {
            if (os == this.OSCompatible)
            {
                return true;
            }
            return false;      
        }
    }
}

File: project_2_java/src/app/FacebookApp.java
package app;

public class FacebookApp implements IAndroidCompatible
{
	private String appName = "Facebook for android";
	private String version = "12.3";

	public final String downloading()
	{
		return "Downloading " + toString();
	}

	public final String getOSCompatible()
	{
		return "Android";
	}

	@Override
	public String toString()
	{
		return "app name: " + appName + ", Version: " + version;
	}

	public final boolean check(String os)
	{
		if (this.getOSCompatible().equals(os))
		{
			return true;
		}
		return false;
	}
}
